homeassistant/components/toon/__init__.py

```diff
@@ -1,289 +1,155 @@
 """Support for Toon van Eneco devices."""
-from functools import partial
+import asyncio
 import logging
-from typing import Any, Dict
 
-from toonapilib import Toon
 import voluptuous as vol
 
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.climate import DOMAIN as CLIMATE_DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import (
     CONF_CLIENT_ID,
     CONF_CLIENT_SECRET,
-    CONF_PASSWORD,
     CONF_SCAN_INTERVAL,
-    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STARTED,
 )
-from homeassistant.core import callback
+from homeassistant.core import CoreState, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
 from homeassistant.helpers import config_validation as cv, device_registry as dr
-from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
-from homeassistant.helpers.entity import Entity
-from homeassistant.helpers.event import async_track_time_interval
-from homeassistant.helpers.typing import ConfigType, HomeAssistantType
-
-from . import config_flow  # noqa: F401
-from .const import (
-    CONF_DISPLAY,
-    CONF_TENANT,
-    DATA_TOON,
-    DATA_TOON_CLIENT,
-    DATA_TOON_CONFIG,
-    DATA_TOON_UPDATED,
-    DEFAULT_SCAN_INTERVAL,
-    DOMAIN,
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    OAuth2Session,
+    async_get_config_entry_implementation,
 )
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_AGREEMENT_ID, DOMAIN
+from .coordinator import ToonDataUpdateCoordinator
+from .oauth2 import register_oauth2_implementations
+
+ENTITY_COMPONENTS = {
+    BINARY_SENSOR_DOMAIN,
+    CLIMATE_DOMAIN,
+    SENSOR_DOMAIN,
+    SWITCH_DOMAIN,
+}
 
 _LOGGER = logging.getLogger(__name__)
 
 # Validation of the user's configuration
 CONFIG_SCHEMA = vol.Schema(
     {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_CLIENT_ID): cv.string,
-                vol.Required(CONF_CLIENT_SECRET): cv.string,
-                vol.Required(
-                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-                ): vol.All(cv.time_period, cv.positive_timedelta),
-            }
+        DOMAIN: vol.All(
+            cv.deprecated(CONF_SCAN_INTERVAL),
+            vol.Schema(
```

Comment:
Shouldn't you still accept `CONF_SCAN_INTERVAL` in your schema or else it will be marked invalid.